chanzuckerberg/czgenepi
# Maya/173849 crud notifs

### Summary
- **What:** 
  - Add notifications for deleting samples
  - Move notifications into the components that actually control them, instead of in data subview
  - I thought I did this before in the cache PR, but make a more sensible name for callbacks on mutations
- **Why:** 
  - CRUD v0
  - To remove a lot of unnecessary state/prop passing between data subview and the components that generate notifications
  - To avoid developer confusion between a components callback and the cache's callback
- **Ticket:** [[sc-173849]](https://app.shortcut.com/genepi/story/173849)
- **Env:** https://notifs-frontend.dev.czgenepi.org [pending]
- **Design:** https://www.figma.com/file/QuQtuWj8iU7nqI6AREd0RE/CRUD-V0?node-id=88%3A64261

### Testing
1. Delete a sample and ensure the notification appears
1. Turn Network throttling in dev tools to `Offline` and try to delete again
2. Error notification should appear
3. Test out other notifications, such as creating a NS tree, creating an Usher tree, downloading offline, etc.

### Notes
Turning off whitespace changes will save you a lot of reading here.

### Checklist
